Using variable only in slice leads to false positive for F841: Locale variable is assigned but never used.

Applying ruff on the following code leads to `repro.py:6:5: F841 Local variable 'shift_by' is assigned to but never used`:
```python
# File repro.py
from __future__ import annotations


def foo() -> None:
    shift_by = 1
    for a in list([1, 2])[shift_by:]:
        print(a)
```

Tested on ruff version `0.0.206`

--- a/resources/test/fixtures/pyflakes/F841_2.py
+++ b/resources/test/fixtures/pyflakes/F841_2.py
@@ -1,0 +1,6 @@
+from __future__ import annotations
+
+# test case for https://github.com/charliermarsh/ruff/issues/1552
+def _():
+    x = 0
+    list()[x:]

--- a/src/pyflakes/mod.rs
+++ b/src/pyflakes/mod.rs
@@ -102,6 +102,7 @@ mod tests {
     #[test_case(CheckCode::F823, Path::new("F823.py"); "F823")]
     #[test_case(CheckCode::F841, Path::new("F841_0.py"); "F841_0")]
     #[test_case(CheckCode::F841, Path::new("F841_1.py"); "F841_1")]
+    #[test_case(CheckCode::F841, Path::new("F841_2.py"); "F841_2")]
     #[test_case(CheckCode::F842, Path::new("F842.py"); "F842")]
     #[test_case(CheckCode::F901, Path::new("F901.py"); "F901")]
     fn checks(check_code: CheckCode, path: &Path) -> Result<()> {

--- a/src/python/typing.rs
+++ b/src/python/typing.rs
@@ -1,6 +1,6 @@
 use once_cell::sync::Lazy;
 use rustc_hash::{FxHashMap, FxHashSet};
-use rustpython_ast::Expr;
+use rustpython_ast::{Expr, ExprKind};
 
 use crate::ast::helpers::{collect_call_paths, dealias_call_path, match_call_path};
 
@@ -219,6 +219,12 @@ pub fn match_annotated_subscript<F>(
 where
     F: Fn(&str) -> bool,
 {
+    if !matches!(
+        expr.node,
+        ExprKind::Name { .. } | ExprKind::Attribute { .. }
+    ) {
+        return None;
+    }
     let call_path = dealias_call_path(collect_call_paths(expr), import_aliases);
     if !call_path.is_empty() {
         for (module, member) in SUBSCRIPTS {

